js/controller.js

```diff
@@ -993,8 +1000,46 @@ OO.plugin("Html5Skin", function (OO, _, $, W) {
         clearTimeout(this.state.timer);
         this.state.timer = null;
       }
+    },
+
+    //determine how aspect ratio is enabled
+    enableAspectRatio: function() {
+      //use fixed aspect ratio number from skinConfig
+      if(this.skin.props.skinConfig.responsive.aspectRatio && this.skin.props.skinConfig.responsive.aspectRatio != "auto") {
+        this.state.mainVideoAspectRatio = this.skin.props.skinConfig.responsive.aspectRatio;
+        this.setAspectRatio(this.state)
+      }
+      //auto detect and update aspect ratio (default)
+      else {
+        if(this.state.isInitialPlay) {
+          this.getIntrinsicDimensions(this.setAspectRatio);
+          this.state.isInitialPlay = false;
+        }
+      }
+    },
+
+    //get original video width/height dimensions
+    getIntrinsicDimensions: function(callback) {
+      var video = this.state.mainVideoElement.get(0);
+      this.state.mainVideoAspectRatio = this.calculateAspectRatio(video.videoWidth, video.videoHeight);
+      if(typeof callback === "function") {
+        callback(this.state);
+      }
+    },
+
+    //returns original video aspect ratio
+    calculateAspectRatio: function(width, height) {
+      var aspectRatio = ((height/width)*100).toFixed(2);
```

Comment:
a little bit of math here.. make sure you are not dividing with 0. Put a safety check on it.